### ISSUE TITLE
Bionic menu should reserve some invlets

### ISSUE THREAD

**Issue by maciekk** [CleverRaven/Cataclysm-DDA#75655]
### Describe the bug

You can assign (and even have autoassigned) invlets to bionic powers which conflicts with menu options (e.g., 's' for sort order for the bionics).

As is, when pressing a letter shared by bionic power and menu option, the menu option wins.

### Attach save file

N/A, this is not game specific

### Steps to reproduce

1. get char w/bionic powers
2. assign 's' to one of the powers (i.e., '=', old letter, 's')
3. press 's'
4. menu sort option dialog comes up, does not activate the bionic power associated with 's'

### Expected behavior

Game should (i) not auto-assign invlet letters which are already "taken" by menu options/commands; (ii) should prevent user from assigning those very same letters, perhaps with a warning (e.g., "that invlet is reserved"); (iii) bionic powers should always activate when their invlet was hit (assuming sufficient power, etc)

### Versions and configuration

- OS: Linux
    - OS Version: Distributor ID: Linuxmint; Description: Linux Mint 21.3; Release: 21.3; Codename: virginia; 
- Game Version: c06f12d [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions],
    Magiclysm [magiclysm]
]

### Additional context

The issue is that get_free_invlet() chooses letters from bionic_chars, and the latter allows all characters, does not reserve any letters used by the menu.

Solution is likely to trim down bionic_chars, or create seperate reserved_chars set which is incorporated during invlet choosing, like it is done in Magiclysm's spell list menu.

**Comment by PatrikLundell** [comment]
Provide a save.
While it's not game specific, setting up the test conditions requires a lot of time (get and install enough bionics to create an assignment clash).

**Comment by PatrikLundell** [comment]
/Confirmed

There is a problem with making assumptions on what others know and thus what's trivial to reproduce. I, for one, know nothing about bionic starts because I've never had any interest in them, and the same goes for mods, as I've only played with one mod once.
I know I can do some things with the debug menu, but other parts are completely unexplored to the extent that I may not even realize it might be present.

A very clear set of instructions on how to set up a situation from a new game (or in any existing game) can work, but it's easy to omit "trivial" steps or just provide vague references because you're so familiar with the sequence yourself.

**Comment by maciekk** [comment]
Yup, fair enough. I kind of assumed all the main contributors had intimate
knowledge of all code, but heh, yeah, I get that the reality is closer to a
rag-tag band of contributors, each with their own areas of expertise (based
on what they like in the game). And indeed, keep forgetting Magiclysm is
not even part of the core experience, even if popular.
